gossip/gossip.go

```diff
@@ -197,6 +223,99 @@ func (g *Gossip) SetNodeDescriptor(desc *roachpb.NodeDescriptor) error {
 	return nil
 }
 
+// SetStallInterval sets the interval between periodic checks to
+// determine whether the node is not fully connected to the gossip
+// network.
+func (g *Gossip) SetStallInterval(interval time.Duration) {
+	g.mu.Lock()
+	defer g.mu.Unlock()
+	g.stallInterval = interval
+}
+
+// SetBootstrapInterval sets a minimum interval between successive
+// attempts to connect to new hosts in order to join the gossip
+// network.
+func (g *Gossip) SetBootstrapInterval(interval time.Duration) {
+	g.mu.Lock()
+	defer g.mu.Unlock()
+	g.bootstrapInterval = interval
+}
+
+// SetCullInterval sets the interval between periodic shutdown of
+// outgoing gossip client connections in an effort to improve the
+// fitness of the network.
+func (g *Gossip) SetCullInterval(interval time.Duration) {
+	g.mu.Lock()
+	defer g.mu.Unlock()
+	g.cullInterval = interval
+}
+
+// SetPersistence provides an instance of the Persistence interface
+// for reading and writing gossip bootstrap data from persistent
+// storage. This should be invoked as early in the lifecycle of a
+// gossip instance as possible, but can be called at any time.
+func (g *Gossip) SetPersistence(persist Persistence) error {
+	// Read the bootstrap info from the persistent store.
+	var storedBI BootstrapInfo
+	err := persist.ReadBootstrapInfo(&storedBI)
+	if err != nil {
+		return err
+	}
+
+	g.mu.Lock()
+	defer g.mu.Unlock()
+	g.persist = persist
+
+	// If the bootstrap info is empty, store any addresses we've become
+	// aware of through the --gossip bootstrap hosts we've managed to
+	// connect to.
+	if len(storedBI.Addresses) == 0 {
```

Comment:
Don't we want to merge the stored values with the `--gossip` flags even if `storedBI` is non-empty? There's no guarantee that any of the old addresses will still be usable (if e.g. the whole cluster was taken offline for a network overhaul)